packages/deployer/internal/process-contracts.js

```diff
@@ -1,60 +1,61 @@
 const fs = require('fs/promises');
-const { getContractAST } = require('@synthetixio/core-js/utils/hardhat/artifacts');
+const { parseFullyQualifiedName } = require('hardhat/utils/contract-names');
 const { getBytecodeHash } = require('@synthetixio/core-js/utils/ethers/contracts');
 const { findInheritedContractNames } = require('@synthetixio/core-js/utils/ast/finders');
-const { contractIsModule, getContractFilePath } = require('./contract-helper');
+const { getContractFilePath } = require('./contract-helper');
 
 /**
  * Initialize contract metadata on hre.deployer.deployment.*
  * This will in turn save all the necessary data to deployments file.
- * @param {string} contractName
+ * @param {string} contractFullyQualifiedName
  */
-async function initContractData(contractName) {
+async function initContractData(contractFullyQualifiedName, extraData = {}) {
   const { deployment, previousDeployment } = hre.deployer;
 
-  const { sourceName, abi, deployedBytecode } = await hre.artifacts.readArtifact(contractName);
+  const { sourceName, contractName } = parseFullyQualifiedName(contractFullyQualifiedName);
+  const { abi, deployedBytecode } = await hre.artifacts.readArtifact(contractFullyQualifiedName);
 
-  const previousData = previousDeployment?.general.contracts[contractName] || {};
-  const isModule = contractIsModule(sourceName);
+  const previousData = previousDeployment?.general.contracts[contractFullyQualifiedName] || {};
   const deployedBytecodeHash = getBytecodeHash(deployedBytecode);
 
-  const generalData = {
+  deployment.general.contracts[contractFullyQualifiedName] = {
     deployedAddress: '',
     deployTransaction: '',
     ...previousData,
+    ...extraData,
     deployedBytecodeHash,
+    contractFullyQualifiedName,
+    contractName,
     sourceName,
   };
 
-  if (isModule) {
-    generalData.isModule = true;
-  }
```

Comment:
This is now setted by the user of the function, using the new `extraData`

Reply:
Cool. Did you check how this affects the CLI tho? Not sure if the CLI was looking at the old format and is broken by this.

Reply:
Yes! it doesn't affect it :)